Set event time and date in firehose script

kdelee opened the pull request:
For testing automation analytics performance, having all these events happen in january does not work. We need them to happen more recently and at different hours.

Going to test this now on a live tower

kdelee committed:
make time delta for firehose events configurable

This was hardcoded to back in january, which make it of limited use testing
automation analytics behavior when there are many events, we need it to be sometime in the past two months.

Will take another pass to do the hours delta

kdelee committed:
add option to set hour delta on firehose

kdelee commented:
tested on a tower box by curl'ing this version of the script over

args passed: awx-python firehose.py --days-delta 1 --hours-delta 2 --jobs 1 --events 3

example event created:
```
        {
            "id": 117927115,
            "type": "job_event",
            "url": "/api/v2/job_events/117927115/",
            "related": {
                "job": "/api/v2/jobs/27853/",
                "children": "/api/v2/job_events/117927115/children/",
                "host": "/api/v2/hosts/1/"
            },
            "summary_fields": {
                "host": {
                    "id": 1,
                    "name": "localhost",
                    "description": ""
                },
                "job": {
                    "id": 27853,
                    "name": "chatty tasks with promptable inv and num messages",
                    "description": "",
                    "status": "successful",
                    "failed": false,
                    "elapsed": 0.0,
                    "type": "job",
                    "job_template_id": 12,
                    "job_template_name": "chatty tasks with promptable inv and num messages"
                },
                "role": {}
            },
            "created": "2020-05-26T20:01:43Z",
            "modified": "2020-05-26T20:01:48Z",
            "job": 27853,
            "event": "playbook_on_start",
            "counter": 3,
            "event_display": "Playbook Started",
            "event_data": {},
            "event_level": 0,
            "failed": false,
            "changed": false,
            "uuid": "f0a62ddb-d8ee-4504-9d9d-048952710677",
            "parent_uuid": "",
            "host": 1,
            "host_name": "localhost",
            "playbook": "hello_world.yml",
            "play": "Example Play",
            "task": "",
            "role": "Hello World",
            "stdout": "X",
            "start_line": 3,
            "end_line": 4,
            "verbosity": 1
        },
```

takeaway:

got these date values the way I want them:

```
            "created": "2020-05-26T20:01:43Z",
            "modified": "2020-05-26T20:01:48Z",
```

ryanpetrello commented on tools/scripts/firehose.py:
```diff
@@ -61,11 +61,15 @@
-    def __init__(self, job_id, rows, *args, **kwargs):
+    def __init__(self, job_id, rows, time_delta, *args, **kwargs):
         self.rows = rows
+        created_time = datetime.datetime.today() - datetime.timedelta(days=time_delta[0], hours=time_delta[1], seconds=5)
```
It might be worth using `timeit` to see what the cost is for running _this_ code 1B times, since that was one of our original use cases here - how much additional timing/overhead does it cost us to generate these strings 1B times?

kdelee replied:
ah I can take that out of there and do it elsewhere

kdelee committed:
Also apply time delta to job

This way they all happened on the same day

kdelee committed:
create timestamp outside loop

this may be expensive

kdelee committed:
be a bit more respectful of event request

minimum events made now is 10,000, used to be 10,000*number of cores.

could dig deeper but this is better for debugging changes to script

kdelee commented:
taking care of 
```
                                "./tools/scripts/firehose.py:61:1: E302 expected 2 blank lines, found 1",
                                "./tools/scripts/firehose.py:159:30: E226 missing whitespace around arithmetic operator",
                                "./tools/scripts/firehose.py:159:56: E226 missing whitespace around arithmetic operator",
                                "./tools/scripts/firehose.py:159:83: E226 missing whitespace around arithmetic operator",
                                "./tools/scripts/firehose.py:159:110: E226 missing whitespace around arithmetic operator",
```

kdelee committed:
fix flake8

kdelee commented:
@ryanpetrello moved generating those strings to a place where it only happens once.

There are still the calculation of `now() - time_delta` in the creation of jobs which I could move if you are worried about it. Granted original script did `now()` for each job as well

kdelee commented:
tested one more time, working for me going to put mergit label on it
